pkg/model/k8s_target.go

```diff
@@ -55,6 +55,15 @@ type K8sTarget struct {
 	// re-evaluated.
 	pathDependencies []string
 	localRepos       []LocalGitRepo
+
+	// HACK(milas): Live Update-only targets don't have an ImageMap because
+	// 	there's no Tilt-managed build despite being modeled as an ImageTarget
+	// 	in the engine. From the API perspective, this is fine: there is not
+	// 	actually a dependency between KubernetesApply<>LiveUpdate API types,
+	// 	but this breaks engine invariants because it expects the build DAG to
```

Comment:
not blocking on this PR but - how hard would it be to skip the DAG check on Kubernetes objects if live-update-v2 is on?

my goal is that as soon as docker-compose is on live-update-v2, all the old live-update-v1 checks will go away

Reply:
That should be trivial - can just check that all the `ImageTarget`s have `LiveUpdateReconciler == true`!

I think we'll still need this hack in `DockerComposeTarget` since it's always v1 reconciler, right? (Hopefully no K8s users are turning it off)

Reply:
(the above diff _should_ be sufficient to replace this PR and work with both v1/v2)

Reply:
ya that's fine too

Reply:
Rebased this PR to use that technique, way less code and easier to get rid of in the future 🗑️